#166001296 Share event on a selected channel

albert85 opened the pull request:
#### What Does This PR Do?
It shares an event on a selected channel

#### Description Of Task To Be Completed
- As a user, when I click on the Share on Slack button on the event details page, and I select a channel, the appropriate request should be made to the backend and the event shared on the specified channel.

#### Any Background Context You Want To Provide?
N/A

#### How can this be manually tested?
- Pull the branch `ft-share-event-166001296`
- Navigate to the dashboard and click on an existing event or create a new event if non exists.
- Click on share button and share on any channel on the dropdown to share the event.

#### What are the relevant pivotal tracker stories?
[#166001296](https://www.pivotaltracker.com/story/show/166001296)

#### Screenshot(s)
<img width="1437" alt="Screenshot 2019-05-16 at 9 50 17 PM" src="https://user-images.githubusercontent.com/28366272/57886236-a7bf5200-7824-11e9-8404-7662669147e9.png">

brianbirir commented:
@albert85 Thanks for implementing this but I'm getting a 400 BAD REQUEST error. On checking the response output via Google Chrome Inspect:

```
{
  "errors": [
    {
      "locations": [
        {
          "column": 19,
          "line": 1
        }
      ],
      "message": "Unknown type \"ShareEventInput\". Perhaps you meant \"CreateEventInput\", \"UpdateEventInput\", \"AttendEventInput\", \"DeactivateEventInput\" or \"SendEventInviteInput\"?"
    },
    {
      "locations": [
        {
          "column": 3,
          "line": 2
        }
      ],
      "message": "Cannot query field \"shareEvent\" on type \"Mutation\". Did you mean \"createEvent\" or \"updateEvent\"?"
    }
  ]
}
```

It's similar to this https://github.com/AndelaOSP/Andela-Socials/pull/190#issuecomment-492942802 but I know @njeri-ngigi fixed it. Is there something that I might have missed on my end?
@brianbirir you have to run `make build` again

albert85 commented:
> @albert85 Thanks for implementing this but I'm getting a 400 BAD REQUEST error. On checking the response output via Google Chrome Inspect:
> 
> ```
> {
>   "errors": [
>     {
>       "locations": [
>         {
>           "column": 19,
>           "line": 1
>         }
>       ],
>       "message": "Unknown type \"ShareEventInput\". Perhaps you meant \"CreateEventInput\", \"UpdateEventInput\", \"AttendEventInput\", \"DeactivateEventInput\" or \"SendEventInviteInput\"?"
>     },
>     {
>       "locations": [
>         {
>           "column": 3,
>           "line": 2
>         }
>       ],
>       "message": "Cannot query field \"shareEvent\" on type \"Mutation\". Did you mean \"createEvent\" or \"updateEvent\"?"
>     }
>   ]
> }
> ```
> 
> It's similar to this [#190 (comment)](https://github.com/AndelaOSP/Andela-Socials/pull/190#issuecomment-492942802) but I know @njeri-ngigi fixed it. Is there something that I might have missed on my end?

olusoladavid commented on client/assets/pages/_event_details-page.scss:
```diff
@@ -249,4 +249,21 @@
+  &__dp-wrapper {
+    list-style-type: none;
+    margin: 0;
+    padding: rem(3px) !important;
```
I think we don't need this padding on the list container. It shows a whitespace around the list items which is not really looking pretty. What we can do here is to set a padding of 0 on the `dp-wrapper` and then on each list item, we can have `padding: rem(5px) rem(12px)`. See screenshot for example:
<img width="596" alt="Screenshot 2019-05-17 at 8 39 20 AM" src="https://user-images.githubusercontent.com/30433120/57911445-b25b0500-787f-11e9-84ef-fcb6d06261f6.png">

olusoladavid commented on client/components/common/DropDownList.js:
```diff
@@ -0,0 +1,35 @@
+import React, { Component } from 'react';
+import PropTypes from 'prop-types';
+import onClickOutside from 'react-onclickoutside';
```
I noticed we didn't end up using this `react-onclickoutside` package? The dropdown is not hidden when we click outside its bounds.

albert85 replied:
resolved

albert85 replied:
resolved

olusoladavid commented on client/components/common/DropDownList.js:
```diff
@@ -0,0 +1,35 @@
+
+    return (
+      <div>
+        <ul className={`${classes}__dp-wrapper`}>
```
About the `classes` prop, I was wondering what the intention of this prop is. In its current form, it does not really allow a person re-using this component to override the styles. For example, if we pass in a custom class like `my-custom-list`, it gets concatenated with `__dp-wrapper` and then no style will be applied to the dropdown. The current setup works because `__dp-wrapper` was defined in the EventDetails page stylesheet. If we want to reuse in a different page, we would have to copy the `__dp-wrapper` styles over to the new page. This leads to code duplication. Could we make the className application more flexible?

albert85 replied:
resolved

olusoladavid reviewed (commented):
Great job on this implementation @albert85. I made some comments on how we can improve it. Please check them out. Cheers 🥂

albert85 committed:
feat(event-details): share event on channel

- refactor codebase
- add a mutation to consume share event endpoint
- update event action to entertain share event
- adjusting dropdown padding
- implement onclickoutside event
- add a scroll view

[Finishes #166001296]

olusoladavid reviewed (approved):
Great job on this implementation @albert85. Works as required.
